sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs

```diff
@@ -2,74 +2,198 @@
         }
 
-        public virtual TokenCredential CreateManagedIdentityCredential(DefaultAzureCredentialOptions options)
+        public virtual TokenCredential CreateManagedIdentityCredential()
         {
             return new ManagedIdentityCredential(new ManagedIdentityClient(
                 new ManagedIdentityClientOptions
                 {
-                    ResourceIdentifier = options.ManagedIdentityResourceId,
-                    ClientId = options.ManagedIdentityClientId,
+                    ResourceIdentifier = Options.ManagedIdentityResourceId,
+                    ClientId = Options.ManagedIdentityClientId,
                     Pipeline = Pipeline,
-                    Options = options,
+                    Options = Options,
                     InitialImdsConnectionTimeout = TimeSpan.FromSeconds(1)
                 })
             );
         }
 
-        public virtual TokenCredential CreateSharedTokenCacheCredential(string tenantId, string username)
+        public virtual TokenCredential CreateSharedTokenCacheCredential()
         {
-            return new SharedTokenCacheCredential(tenantId, username, null, Pipeline);
+            return new SharedTokenCacheCredential(Options.SharedTokenCacheTenantId, Options.SharedTokenCacheUsername, Options, Pipeline);
         }
 
-        public virtual TokenCredential CreateInteractiveBrowserCredential(string tenantId, string clientId)
+        public virtual TokenCredential CreateInteractiveBrowserCredential()
         {
+            var options = new InteractiveBrowserCredentialOptions
+            {
+                TokenCachePersistenceOptions = new TokenCachePersistenceOptions(),
+                AuthorityHost = Options.AuthorityHost,
+                TenantId = Options.InteractiveBrowserTenantId
+            };
+
+            foreach (var addlTenant in Options.AdditionallyAllowedTenants)
+            {
+                options.AdditionallyAllowedTenants.Add(addlTenant);
+            }
+
             return new InteractiveBrowserCredential(
-                tenantId,
-                clientId ?? Constants.DeveloperSignOnClientId,
-                new InteractiveBrowserCredentialOptions { TokenCachePersistenceOptions = new TokenCachePersistenceOptions() },
+                Options.InteractiveBrowserTenantId,
+                Options.InteractiveBrowserCredentialClientId ?? Constants.DeveloperSignOnClientId,
+                options,
                 Pipeline);
         }
 
-        public virtual TokenCredential CreateAzureCliCredential(TimeSpan? cliProcessTimeout)
+        public virtual TokenCredential CreateAzureCliCredential()
         {
-            return new AzureCliCredential(Pipeline, default, new AzureCliCredentialOptions() { CliProcessTimeout = cliProcessTimeout});
+            var options = new AzureCliCredentialOptions
+            {
+                TenantId = Options.TenantId,
+				CliProcessTimeout = Options.DeveloperCredentialTimeout
```

Comment:
nit: weird spacing here and a few other places

Reply:
fixed.